src/Sarif.Converters/TSLintConverter.cs

```diff
@@ -0,0 +1,147 @@
+using System.Runtime.CompilerServices;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis.Sarif.Writers;
+
+[assembly: InternalsVisibleTo("Sarif.Driver.UnitTests.dll,PublicKey=0024000004800000940000000602000000240000525341310004000001000100433fbf156abe971"+
+    "8142bdbd48a440e779a1b708fd21486ee0ae536f4c548edf8a7185c1e3ac89ceef76c15b8cc2497906798779a59402f9b9e27281fb15e7111566cdc9a9f8326301d45320623c52"+
+    "22089cf4d0013f365ae729fb0a9c9d15138042825cd511a0f3d4887a7b92f4c2749f81b410813d297b73244cf64995effb1")]
+namespace Microsoft.CodeAnalysis.Sarif.Converters
+{
+    public class TSLintConverter : ToolFileConverterBase
+    {
+        private readonly ITSLintLoader loader;
+        
+        public TSLintConverter()
+        {
+            loader = new TSLintLoader();
+        }
+
+        public TSLintConverter(ITSLintLoader loader)
+        {
+            this.loader = loader ?? throw new ArgumentNullException(nameof(loader));
+        }
+        
+        public override void Convert(Stream input, IResultLogWriter output, LoggingOptions loggingOptions)
+        {
+            input = input ?? throw new ArgumentNullException(nameof(input));
+
+            output = output ?? throw new ArgumentNullException(nameof(output));
+
+            output.Initialize(id: null, automationId: null);
+
+            TSLintLog tsLintLog = loader.ReadLog(input);
+
+            Tool tool = new Tool
+            {
+                Name = "TSLint"
+            };
+
+            output.WriteTool(tool);
+
+            var results = new List<Result>();
+            foreach(TSLintLogEntry entry in tsLintLog)
+            {
+                results.Add(CreateResult(entry));
+            }
+
+            var fileInfoFactory = new FileInfoFactory(MimeType.DetermineFromFileExtension, loggingOptions);
+            Dictionary<string, FileData> fileDictionary = fileInfoFactory.Create(results);
+            
+            if (fileDictionary?.Any() == true)
+            {
+                output.WriteFiles(fileDictionary);
+            }
+
+            output.OpenResults();
+            output.WriteResults(results);
+            output.CloseResults();
+        }
+
+        internal Result CreateResult(TSLintLogEntry tslEntry)
+        {
+            tslEntry = tslEntry ?? throw new ArgumentNullException(nameof(tslEntry));
+
+            Result result = new Result()
+            {
+                RuleId = tslEntry.RuleName,
+                Message = tslEntry.Failure
+            };
+
+            switch (tslEntry.RuleSeverity)
+            {
+                case "ERROR":
+                    result.Level = ResultLevel.Error;
+                    break;
+                case "WARN":
+                case "WARNING":
+                    result.Level = ResultLevel.Warning;
+                    break;
+                case "DEFAULT":
+                    result.Level = ResultLevel.Default;
+                    break;
```

Comment:
What are the semantics when a TSLint log entry has a `RuleSeverity` of `"DEFAULT"`? It might not match what SARIF's `ResultLevel.Default` means.

Reply:
I have no idea, I couldn't find a spec. I can remove this case if you prefer.

Reply:
@boAndron Lacking any other information, I propose to use `ResultLevel.Note` for any TSLint `RuleSeverity` other than `"ERROR"`, `"WARN"`, or `"WARNING"`. In SARIF, that means "a purely informational log entry".

@michaelcfanning I'm going to push (and merge) that change, but feel free to chime in if you disagree.